doomed_patient/doomed_patient/Player.cpp

```diff
@@ -5,11 +5,21 @@
 
 Player::Player(PatientGame* game, VectorXY startCoordinates)
 	: Character(game, startCoordinates, game->getPlayerSprite()),
-	speed(2)
+	speed(0)
 {
+	currentCell = game->level.getCell(VectorXY(0, 0));
+	centre = currentCell->getCentre();
 }
 
 void Player::move(Directions::Direction direction)
 {
-	centre += Directions::getDirectionVector(direction) * speed;
+	
+	// Put collision detection here
+
+	if (currentCell->getEdge(direction)->isWall())
+	{
+		return;
+	}
+	currentCell = game->level.getCell(currentCell->getCoordinates() + Directions::getDirectionVector(direction));
+	centre = currentCell->getCentre();
```

Comment:
You want to put ```centre += Directions::getDirectionVector(direction) * speed;``` back in to make the player move at a sensible rate, and set speed back to 2 or whatever number